webxrlayers-1.bs

```diff
@@ -275,11 +275,17 @@ enable optical chromatic aberration correction for the layer. Only valid if this
 ISSUE: describe what chromaticAberrationCorrection does.
 
 The <dfn attribute for="XRCompositionLayer">needsRedraw</dfn> attribute signals that the {{XRCompositionLayer}} should be
-rerendered in the next [=XR animation frame=].
+rerendered in the next [=XR animation frame=]. It MAY be set when [=the underlying resources of a layer are reallocated=] or
+when the [=XR Compositor=] can no longer reproject the layer. Failing to redraw the content in the next [=XR animation frame=]
+might cause flickering or other side effects.
 
-NOTE: the user agent might set this flag if the underlying resources were reallocated or if the [=XR Compositor=] can no
-longer reproject the layer. Failing to redraw the content in the next [=XR animation frame=] might cause flickering or
-other side effects.
+<div class="algorithm" data-algorithm="redrawLayerAlgo">
+When <dfn>the underlying resources of a layer are reallocated</dfn> for an {{XRCompositionLayer}} |layer|,
+the user agent MUST run the following steps:
+  1. Set the |layer|'s {{XRCompositionLayer/needsRedraw}} array to an empty array.
+  1. If |layer| is not a {{XRProjectionLayer}}, [=queue a task=] to [=fire an event=] named {{redraw}} on |layer|.
```

Comment:
I like the omission of `onredraw` from the projection layer specifically, because we really just want devs to redraw that all the time anyway. 👍